test/e2e/middleware-custom-matchers-basepath/app/middleware.js

```diff
@@ -0,0 +1,19 @@
+import { NextResponse } from 'next/server'
+
+export default function middleware(request) {
+  const res = NextResponse.rewrite(new URL('/', request.url))
```

Comment:
Why is `new URL()` being used here? This breaks the automatic `basePath` handling for rewrites so this fails when testing against a deployment should use `request.nextUrl.pathname = '/'` and pass `request.nextUrl` instead. 


Also the generated regex looks off as the `basePath` should only be present once but is there twice:
`^/docs(?:/(docs/_next/data/[^/]{1,}))?/hello(.json)?[/#\\?]?$`

Reply:
Fixed the first one. About the regex, how did you get the value? As far as I checked, it is `^\\/docs(?:\\/(_next\\/data\\/[^/]{1,}))?\\/hello(.json)?[\\/#\\?]?$`

Reply:
Seems this happens only on vercel deployed and it's an existing problem before the PR, will fix anyway.